test/jdk/java/awt/EventDispatchThread/InterruptEDTTest.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @key headful
+ * @bug 7184401
+ * @summary Verify that events are not lost when EDT is interrupted.
+ * @modules java.desktop/sun.awt
+ * @run main InterruptEDTTest
+ */
+
+public class InterruptEDTTest {
+
+    private static Frame frame = null;
+    private static volatile Thread edt = null;
+    private static volatile Robot robot = null;
+    private static volatile int xLocation;
+    private static volatile int yLocation;
+    private static volatile int width;
+    private static volatile int height;
+
+    public static void main(String args[]) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> {
+                edt = Thread.currentThread();
+                frame = new Frame("Frame");
+                frame.addMouseListener(new MouseAdapter() {
+                    public void mouseClicked(MouseEvent me) {
+                        System.out.println("Mouse clicked Event: " + me);
+                    }
+                });
+                frame.setBounds(350, 50, 400, 400);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+            });
+            ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+            EventQueue.invokeAndWait(() -> {
+                xLocation = frame.getX();
+                yLocation = frame.getY();
+                width = frame.getWidth();
+                height = frame.getHeight();
+            });
+            ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+            EventQueue.invokeLater(() -> {
+                robot.mouseMove(xLocation + width / 2, yLocation + height / 2);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        edt.interrupt();
+        ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+        try {
+            EventQueue.invokeAndWait(() -> {
+                xLocation = frame.getX();
+                yLocation = frame.getY();
+                width = frame.getWidth();
+                height = frame.getHeight();
+            });
+            robot.mouseMove(xLocation + width / 3, yLocation + height / 3);
```

Comment:
Line# 75 robot.mouseMove is called with in EDT? but where as here its not. Is there any reason for not invoking this in EDT ?

Reply:
Yes.  Lie #75 on EDT via the invokeLater call , followed by the interrupt() call on the EDT is intentional. This mimics the Netbeans actions which caused a failure
1.Widgets are created on EDT; (Frame creation)
2. another code posts some events to them on EDT(Robot actions on Line 75 on EDT via the invokeLater)
 From another thread some code calls explicitly  edt.interrupt().
3. Above steps causes a hang on jdk7u6 build, while the later builds it is passing. 
4. Line# 92 is simply an additional validation of the code not being hanged until now and proceeding ahead in the flow.